hermes-console-vue/src/composables/search/useSearch.ts

```diff
@@ -0,0 +1,92 @@
+import { querySubscriptions, queryTopics } from '@/api/hermes-client';
+import { ref } from 'vue';
+import type { Ref } from 'vue';
+import type { Subscription } from '@/api/subscription';
+import type { Topic } from '@/api/topic';
+
+export interface UseSearch {
+  topics: Ref<Topic[] | undefined>;
+  subscriptions: Ref<Subscription[] | undefined>;
+  queryTopicsFn: (filter: SearchFilter, pattern: string) => void;
+  querySubscriptionsFn: (filter: SearchFilter, pattern: string) => void;
+  loading: Ref<boolean>;
+  error: Ref<UseSearchErrors>;
+}
+
+export interface UseSearchErrors {
+  fetchError: Error | null;
```

Comment:
This should be extracted to two variables: queryTopics and querySubscriptions.

Reply:
extracted